### PR TITLE
mac-sequence: Set destination sequence number (DSN) in mac-sequence.c

### PR DESCRIPTION
**Pull request by kkrentz**
The motivation behind this PR is to eliminate duplicate code and to avoid special handling for DSN=0.

### CODE REVIEW

**Comment by arurke** [comment]
@kkrentz In framer-802154.c there was efforts to ensure the sequence number was not zero - which this PR removes. I can't really tell if it was just to check if a new sequence number was needed, or if there is actually some other parts of the mac or lower layers that rely on this behavior. Any thoughts?

**Comment by kkrentz** [comment]
@arurke That code in `framer-802154.c` was necessary because there was no clear responsibility for setting the destination sequence number (DSN). If DSN=0, `framer-802154.c` took over. This PR makes the MAC driver responsible for setting the DSN, obviating the special handling for DSN=0. Also, in line with IEEE 802.15.4, outgoing frames no longer skip over DSN=0.

File: os/net/mac/mac-sequence.h [hunk 1]
```diff
@@ -45,6 +45,16 @@
 #ifndef MAC_SEQUENCE_H
 #define MAC_SEQUENCE_H
 
+/**
+ * brief       Initializes.
```
Comment (arurke) [inline os/net/mac/mac-sequence.h:49]: * `\brief`
* How about "Initializes the MAC sequence number to a random value"?  Making it clear that calling it twice actually influences the value.

**Review by arurke — CHANGES_REQUESTED** [review]

**Review by arurke — APPROVED** [review]

**Comment by arurke** [comment]
Thanks for the explanation and fixes @kkrentz

**Review by nfi — APPROVED** [review]

**Pull request merged**